### PR TITLE
HPCC-16931 Avoid loading dlls when extracting resources in esp

### PR DESCRIPTION
**Pull request by ghalliday**
Signed-off-by: Gavin Halliday <gavin.halliday@lexisnexis.com>

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [x] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [x] This change fixes the problem, not just the symptom
  - [x] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately

## Testing:
I have built the system, and added tracing to ensure the new function was being called and returning the correct entries.
I have not tested properly with anything other than the default manifest, because I don't really know how to.

### CODE REVIEW

**Comment by ghalliday** [comment]
@GordonSmith I'm not sure how to test this properly.  Do you have any suggestions?
@wangkx please take a look at the changes - do they look sensible.

**Comment by GordonSmith** [comment]
@ghalliday If you install the visualizer bundle and submit some its self tests it will add some resources for you.

**Comment by ghalliday** [comment]
Sounds like something else to add to OBT..

File: common/dllserver/thorplugin.cpp [hunk 1]
```diff
@@ -379,6 +233,207 @@ extern bool getResourceFromFile(const char *filename, MemoryBuffer &data, const
...
+
+bool HelperDll::load(bool isGlobal, bool raiseOnError)
+{
+    if (!so.load(name, isGlobal, raiseOnError))
+        return false;
+    return true;
+}
+
+bool HelperDll::loadResources()
+{
```
Comment (wangkx) [inline common/dllserver/thorplugin.cpp]: Do you want to check if (so.loaded())?

**Review by wangkx — COMMENTED** [review]
@ghalliday Other than a question, look sensible.

File: common/dllserver/thorplugin.cpp (hunk 1 above)
Comment (ghalliday) [inline common/dllserver/thorplugin.cpp]: It is a counterpart to load() and locadCurrentExecutable() - neither of which check if it is already loaded.  It isn't a public function, so can't be called more than once.

**Comment by ghalliday** [comment]
@richardkchapman please review/merge.  I've repushed the commit to fix the commit message.

**Comment by richardkchapman** [comment]
@ghalliday The commit message doesn't really describe the change very well for changelog purposes...

**Pull request merged**